cluster-service/deploy/openshift-templates/arohcp-service-template.yml

```
@@ -368,6 +368,107 @@ objects:
+- apiVersion: v1
+  kind: ConfigMap
+  metadata:
+    name: azure-operators-managed-identities-config
+    namespace: ${NAMESPACE}
+  data:
+    azure-operators-managed-identities-config.yaml: |
+      controlPlaneOperatorsIdentities:
+
+        cloud-controller-manager:
+          minOpenShiftVersion: 4.17
+          azureRoleDefinitionResourceId: '/subscriptions/1d3378d3-5a3f-4712-85a1-2485495dfc4b/providers/Microsoft.Authorization/roleDefinitions/ebe170ec-1247-536a-86d9-74c829dd9844'
+          azureRoleDefinitionName: 'Azure Red Hat OpenShift Cloud Controller Manager - Dev'
+          optional: false
+
+        ingress:
+          minOpenShiftVersion: 4.17
+          azureRoleDefinitionResourceId: '/subscriptions/1d3378d3-5a3f-4712-85a1-2485495dfc4b/providers/Microsoft.Authorization/roleDefinitions/589ca160-4fac-501e-ad6c-006a19583727'
+          azureRoleDefinitionName: 'Azure Red Hat OpenShift Cluster Ingress Operator - Dev'
+          optional: false
+
+        disk-csi-driver:
+          minOpenShiftVersion: 4.17
+          azureRoleDefinitionResourceId: '/subscriptions/1d3378d3-5a3f-4712-85a1-2485495dfc4b/providers/Microsoft.Authorization/roleDefinitions/4367fe74-0b43-5033-b629-15d9f28415ac'
+          azureRoleDefinitionName: 'Azure Red Hat OpenShift Disk Storage Operator - Dev'
+          optional: false
+
+        file-csi-driver:
+          minOpenShiftVersion: 4.17
+          azureRoleDefinitionResourceId: '/subscriptions/1d3378d3-5a3f-4712-85a1-2485495dfc4b/providers/Microsoft.Authorization/roleDefinitions/fdc0aaaa-1c3e-548e-ad27-0321e5fab18b'
+          azureRoleDefinitionName: 'Azure Red Hat OpenShift File Storage Operator - Dev'
+          optional: false
+
+        image-registry:
+          minOpenShiftVersion: 4.17
+          azureRoleDefinitionResourceId: '/subscriptions/1d3378d3-5a3f-4712-85a1-2485495dfc4b/providers/Microsoft.Authorization/roleDefinitions/357b9263-656f-5d45-9d7a-ccb825f0683f'
+          azureRoleDefinitionName: 'Azure Red Hat OpenShift Image Registry Operator - Dev'
+          optional: false
+
+        cloud-network-config:
+          minOpenShiftVersion: 4.17
+          azureRoleDefinitionResourceId: '/subscriptions/1d3378d3-5a3f-4712-85a1-2485495dfc4b/providers/Microsoft.Authorization/roleDefinitions/4e4f23fe-3fab-568b-a001-10b233b0f840'
+          azureRoleDefinitionName: 'Azure Red Hat OpenShift Network Operator - Dev'
+          optional: false
+
+      dataPlaneOperatorsIdentities:
+
+        disk-csi-driver:
+          minOpenShiftVersion: 4.17
+          azureRoleDefinitionResourceId: '/subscriptions/1d3378d3-5a3f-4712-85a1-2485495dfc4b/providers/Microsoft.Authorization/roleDefinitions/4367fe74-0b43-5033-b629-15d9f28415ac'
+          azureRoleDefinitionName: 'Azure Red Hat OpenShift Disk Storage Operator - Dev'
+          k8sServiceAccounts:
+            - name:  'system:serviceaccount:openshift-cluster-csi-drivers:azure-disk-csi-driver-operator'
```

I think this should just be the name of the service account. 
```suggestion
            - name:  'azure-disk-csi-driver-operator'
```

When CS is performing the federation, it'll construct the subject as expected, using the name and the namespace
```go
fmt.Sprintf("system:serviceaccount:%s:%s",namespace,name)
```

@bennerv can you confirm?

---

These should always be hardcoded.  I'm actually wondering if CS needs namespace as a field in this config and instead we can just list the full `system:serviceaccount:<namespace>:<service-account-name>` as the config and call it good.  

CS configures these on the hosted cluster CR, so technically it doesn't need to know what namespaces correspond to what.  Dealers choice @machi1990 , but IMO i'd either remove namespace from this config and keep the "full" path, or just use the name of the service account.

---

@bennerv @tony-schndr  We could've given the full `system:serviceaccount:<namespace>:<service-account-name>` and call it good but the way the config was designed is that `name` and `namespace` are to be given (ARO-7364) and CS would use use that info to construct the federation subject name among other. And config loading mechanism in CS has been implemented that way so I would suggest to keep it at it and not change it. 
Meaning that here we should give
```yaml
name: <service-account-name>
namespace: <service-account-namespace>
```
and not the full form for the name.